renderdoc/core/android.cpp

```diff
@@ -1188,18 +1257,28 @@ extern "C" RENDERDOC_API void RENDERDOC_CC RENDERDOC_CheckAndroidPackage(const c
   *flags = AndroidFlags::NoFlags;
 
   bool found = false;
+  string layerPath = "";
 
-  // First, see if the application contains the layer
-  if(SearchForAndroidLayer(deviceID, pkgPath, layerName))
+  // Check a debug location only usable by rooted devices, overriding app's layer
+  if(SearchForAndroidLayer(deviceID, "/data/local/debug/vulkan", layerName, layerPath))
     found = true;
 
-  // Next, check a debug location only usable by rooted devices
-  if(!found && SearchForAndroidLayer(deviceID, "/data/local/debug/vulkan", layerName))
+  // See if the application contains the layer
+  if(!found && SearchForAndroidLayer(deviceID, pkgPath, layerName, layerPath))
     found = true;
 
   // TODO: Add any future layer locations
 
-  if(!found)
+  if(found)
+  {
+    // Check the version of the layer found
```

Comment:
This isn't necessarily set in stone so if you feel really strongly I could be persuaded, but I don't think I want a layer version mismatch to cause a warning in release builds. For release, all layer versions should be backwards compatible (new host and old embedded layer), and anything that breaks that without a very _very_ good reason I'd consider a bug.

So I think maybe this code should be inside `#if ENABLED(RDOC_DEVEL)` so that we can catch the case where someone developing on renderdoc is rapidly iterating and wants to know that their code isn't up to date.

As a separate thing, it would be nice to enhance the capture connection window to alert the user if the connected target was running older code - that way it's a 'soft' failure, in that the capture can still happen but the user is alerted that the target layer is out of date and might be missing bugfixes/features.

Reply:
I"m fine with moving this behind a developer check.  I'm not super comfortable with how tight the coupling becomes, nor adding it so close to a release candidate.